### DISCUSSION TITLE
Make Dfx build command build canisters in parallel

### DISCUSSION THREAD

**Discussion by ByronBecker** [dfinity/sdk#3857] [Feature Requests]
For larger projects, it can take on the order of minutes to build a larger canister. Projects with 7-10 canisters then take a long time to build, especially if one wants to spin up a local environment for testing or frontend development.

It would be much faster if running `dfx build` would build all canisters in the project in parallel.

**Comment by chenyan-dfinity** [comment]
I encourage you to try https://github.com/chenyan2002/mops-cli. Just run `mops-cli build main.mo` to build the canister. If you want to parallelize it, you can put them in a bash script.

**Comment by ByronBecker** [comment]
@nathanosdev, are there any plans to tackle build parallelization? 🙏
